secure/json-rpc/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! The purpose of the LightweightJsonClient presented here is to provide a lightweight and secure
+//! JSON RPC client to talk to the JSON RPC service offered by Libra Full Nodes. This is useful
+//! for various security-critical components (e.g., the secure key manager), as it allows
+//! interaction with the Libra blockchain in a minimal and secure manner.
+//!
+//! Note: While a JSON RPC client implementation already exists in the Libra codebase, this
+//! provides a simpler and (hopefully) more secure implementation with fewer dependencies.
+#![forbid(unsafe_code)]
+
+use crate::error::Error;
+use libra_types::{
+    account_address::AccountAddress, account_state::AccountState,
+    account_state_blob::AccountStateBlob, transaction::SignedTransaction,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use std::convert::TryFrom;
+use ureq::Response;
+
+mod error;
+#[cfg(test)]
+mod tests;
+
+/// Various constants for the JSON RPC client implementation
+const JSON_RPC_VERSION: &str = "2.0";
+const REQUEST_TIMEOUT: u64 = 10_000;
+
+/// This defines a simple and generic interface by which to interact with the JSON RPC service
+/// offered by each Libra Full Node.
+pub trait JsonRpcClient {
+    /// Submits a signed transaction to the Libra blockchain via the JSON RPC API.
+    fn submit_signed_transaction(&self, signed_transaction: SignedTransaction)
+        -> Result<(), Error>;
+
+    /// Returns the associated AccountState for a specific account at a given version height
+    /// using the JSON RCP API.
+    fn get_account_state(
+        &self,
+        account: AccountAddress,
+        version: u64,
+    ) -> Result<AccountState, Error>;
+}
+
+/// Provides a lightweight JsonRpcClient implementation.
+struct LightweightJsonClient {
```

Comment:
I don't think we need the additional naming here... it is already in a different namespace... Simply JsonRpcClient would do.